commit fa2518718e084872963e0e4a7c0f4aba684feed8
Author: Eugene Rudoy <gene.devel@gmail.com>
Date:   Mon Jan 22 22:36:37 2018 +0100

    nandwrite: fix MTD_FILE_MODE_RAW related build problem with kernels < 3.2.0

    miscutils/nandwrite.c: In function 'nandwrite_main':
    miscutils/nandwrite.c:151:35: error: 'MTD_FILE_MODE_RAW' undeclared (first use in this function)
       xioctl(fd, MTDFILEMODE, (void *)MTD_FILE_MODE_RAW);
                                       ^
    Signed-off-by: Eugene Rudoy <gene.devel@gmail.com>

--- miscutils/nandwrite.c
+++ miscutils/nandwrite.c
@@ -60,6 +60,18 @@
 #endif
 
 
+#include <linux/version.h>
+/*
+ * MTD_FILE_MODE_RAW is an enum (i.e. cannot be tested using #ifdef) and is available since kernel 3.2
+ * s. http://elixir.free-electrons.com/linux/v3.2/ident/MTD_FILE_MODE_RAW
+ *
+ * In versions before 3.2 it's called MTD_MODE_RAW
+ * s. http://elixir.free-electrons.com/linux/v3.1/ident/MTD_MODE_RAW
+ */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,2,0)
+#define MTD_FILE_MODE_RAW MTD_MODE_RAW
+#endif
+
 #define IS_NANDDUMP  (ENABLE_NANDDUMP && (!ENABLE_NANDWRITE || (applet_name[4] == 'd')))
 #define IS_NANDWRITE (ENABLE_NANDWRITE && (!ENABLE_NANDDUMP || (applet_name[4] != 'd')))
 
